packages/@ember/-internals/glimmer/tests/integration/helpers/invoke-helper-test.js

```diff
@@ -0,0 +1,618 @@
+
+        runTask(() => destroy(context));
+
+        assert.ok(isDestroyed(context), 'context destroyed');
+        assert.ok(isDestroyed(cache), 'cache destroyed');
+        assert.ok(isDestroyed(instance), 'instance destroyed');
+
+        assert.verifySteps(['instance', 'cache', 'context'], 'destructors ran in correct order');
+      }
+
+      '@test helper destroys correctly when helper cache is destroyed'(assert) {
+        let context = {};
+        let instance;
+
+        class TestHelper extends Helper {
+          constructor() {
+            super(...arguments);
+            instance = this;
+          }
+        }
+
+        let cache = invokeHelper(context, TestHelper);
+
+        registerDestructor(context, () => assert.step('context'));
+        registerDestructor(cache, () => assert.step('cache'));
+        registerDestructor(instance, () => assert.step('instance'));
+
+        runTask(() => destroy(cache));
+
+        assert.notOk(isDestroyed(context), 'context NOT destroyed');
+        assert.ok(isDestroyed(cache), 'cache destroyed');
+        assert.ok(isDestroyed(instance), 'instance destroyed');
+
+        assert.verifySteps(['instance', 'cache'], 'destructors ran in correct order');
+      }
+
+      '@test simple helper destroys correctly when context object is destroyed'(assert) {
+        let context = {};
+
+        let TestHelper = helper(() => {});
+        let cache = invokeHelper(context, TestHelper);
+
+        registerDestructor(context, () => assert.step('context'));
+        registerDestructor(cache, () => assert.step('cache'));
+
+        runTask(() => destroy(context));
+
+        assert.ok(isDestroyed(context), 'context destroyed');
+        assert.ok(isDestroyed(cache), 'cache destroyed');
+
+        assert.verifySteps(['cache', 'context'], 'destructors ran in correct order');
+      }
+
+      '@test throws an error if value is accessed after it is destroyed'() {
+        expectAssertion(() => {
+          let helper = invokeHelper({}, class extends Helper {});
+
+          runTask(() => destroy(helper));
+
+          getValue(helper);
+        }, /You attempted to get the value of a helper after the helper was destroyed, which is not allowed/);
+      }
+
+      '@test asserts if no context object is passed'() {
+        expectAssertion(() => {
+          invokeHelper(undefined, class extends Helper {});
+        }, /Expected a context object to be passed as the first parameter to invokeHelper, got undefined/);
+      }
+
+      '@test asserts if no manager exists for the helper definition'() {
+        expectAssertion(() => {
+          invokeHelper({}, class {});
+        }, /Expected a helper definition to be passed as the second parameter to invokeHelper, but no helper manager was found. The definition value that was passed was `\(unknown function\)`. Did you use setHelperManager to associate a helper manager with this value?/);
+      }
+    }
+  );
+
```

Comment:
I think this needs to be:

```suggestion
  if (
    EMBER_GLIMMER_HELPER_MANAGER &&
    EMBER_GLIMMER_INVOKE_HELPER
  ) {
```

Specifically, if we enable one before the other (which almost always does happen) these tests may fail. As far as I can tell **both** features are required for these tests...

Reply:
probably hard to see but it was nested in the other `if`, I can see how that's confusing though. Updated to check both 👍

Reply:
Ah, gotcha. The indentation was hard to spot on mobile for sure.